src/main/groovy/net/minecraftforge/gradleutils/changelog/ChangelogUtils.groovy

```diff
@@ -4,622 +4,136 @@
             }
         }
     }
 
     private static void setupChangelogGenerationForAllPublications(Project project) {
-        var ext = project.getExtensions().findByName("publishing") as PublishingExtension
-        if (ext == null)
-            return
+        var ext = project.extensions.findByName(PublishingExtension.NAME) as PublishingExtension
+        if (ext == null) return
 
-        //Get each extension and add the publishing task as a publishing artifact.
+        // Get each extension and add the publishing task as a publishing artifact
         ext.publications.withType(MavenPublication).configureEach {
-            //Add the task as a publishing artifact.
             setupChangelogGenerationForPublishing(project, it)
         }
     }
 
     private static ChangelogExtension findParent(Project project) {
         var ext = project.extensions.findByType(ChangelogExtension)
-        if (ext?.task != null)
-            return ext
+        if (ext?.task != null) return ext
 
-        var parent = project.getParent() == project ? null : project.getParent()
+        var parent = project.parent == project ? null : project.parent
         return parent == null ? null : findParent(parent)
     }
 
     /**
      * The recommended way to share task outputs across projects is to export them as dependencies
-     * So for any project that doesn't generate the changelog directly, we must create a copy task
-     * and new configuration
+     * <p>
+     * So for any project that doesn't generate the changelog directly, we must create a
+     * {@linkplain CopyChangelog copy task} and new configuration
      */
-    private static TaskProvider<Task> findChangelogTask(final Project project) {
+    private static TaskProvider<? extends Task> findChangelogTask(Project project) {
         // See if we've already made the task
-        if (project.tasks.names.contains(TASK_NAME))
-            return project.tasks.named(TASK_NAME)
+        if (project.tasks.names.contains(GenerateChangelog.NAME))
+            return project.tasks.named(GenerateChangelog.NAME)
 
-        if (project.tasks.names.contains(COPY_TASK_NAME))
-            return project.tasks.named(COPY_TASK_NAME)
+        if (project.tasks.names.contains(CopyChangelog.NAME))
+            return project.tasks.named(CopyChangelog.NAME)
 
         // See if there is any parent with a changelog configured
         var parent = findParent(project)
-        if (parent == null)
-            return null
+        if (parent == null) return null
 
-        var cfg = project.configurations.register(COPY_TASK_NAME) {
-            canBeConsumed = false
-        }
-
-        project.dependencies.add(COPY_TASK_NAME, project.dependencies.project('path': parent.project.path, 'configuration': TASK_NAME))
+        project.dependencies.add(CopyChangelog.NAME, project.dependencies.project('path': parent.project.path, 'configuration': GenerateChangelog.NAME))
```

Comment:
This should be fine, but in the future probably worth changing this to a detached configuration so that we don't expose it to consumers and it doesn't take up a name in the configuration list.

Reply:
I'll add this as a TODO comment, but since it might break buildscript that use it, it's fine for now.

Reply:
I decided to just go ahead and change it to the detached configuration anyway. Let me know if I should revert that, though. I don't foresee it breaking anything.

```groovy
project.tasks.register(CopyChangelog.NAME, CopyChangelog) {
    var dependency = project.dependencies.project('path': parent.project.path, 'configuration': GenerateChangelog.NAME)
    it.configuration.set project.configurations.detachedConfiguration(dependency).tap { it.canBeConsumed = false }
}
```